tests/types.test.tsx

```diff
@@ -188,3 +194,42 @@ it('state is covariant', () => {
     baz: string
   }> = store
 })
+
+it('Shows UnknownBecauseOfGenericMutators type when dealing with generic mutators', () => {
+  interface State {
+    count: number
+    increment: () => void
+  }
+
+  const foo: <M extends [StoreMutatorIdentifier, unknown][]>() => StateCreator<
+    State,
+    M
+  > = () => (set, get) => ({
+    count: 0,
+    increment: () => {
+      // @ts-expect-errors
+      set({ count: get().count + 1 })
+    },
+  })
+
+  create<State>()(persist(foo()))
+})
+
+it('StateCreator subtyping', () => {
+  interface State {
+    count: number
+    increment: () => void
+  }
+
+  const foo: () => StateCreator<State, []> = () => (set, get) => ({
+    count: 0,
+    increment: () => {
+      set({ count: get().count + 1 })
+    },
+  })
+
+  create<State>()(persist(foo()))
```

Comment:
So fixing the `StateCreator` subtyping no longer errors on this line as it does today... https://tsplay.dev/mq3pjw